src-java/floodlightrouter-topology/floodlightrouter-storm-topology/src/main/java/org/openkilda/wfm/topology/floodlightrouter/FloodlightRouterTopology.java

```diff
@@ -41,500 +48,290 @@
  * Floodlight topology.
  */
 public class FloodlightRouterTopology extends AbstractTopology<FloodlightRouterTopologyConfig> {
+    private final Set<String> regions;
+
+    private final KafkaTopicsConfig kafkaTopics;
     private final PersistenceManager persistenceManager;
 
-    public FloodlightRouterTopology(LaunchEnvironment env) {
+    public FloodlightRouterTopology(LaunchEnvironment env) throws ConfigurationException {
         super(env, FloodlightRouterTopologyConfig.class);
 
-        persistenceManager = PersistenceProvider.getInstance().createPersistenceManager(configurationProvider);
-    }
-
-    private void createKildaFlowHsSpout(TopologyBuilder builder, int parallelism, List<String> kildaFlowTopics) {
-        KafkaSpout kildaFlowHsSpout = buildKafkaSpoutForAbstractMessage(kildaFlowTopics,
-                ComponentType.KILDA_FLOW_HS_KAFKA_SPOUT);
-        builder.setSpout(ComponentType.KILDA_FLOW_HS_KAFKA_SPOUT, kildaFlowHsSpout, parallelism);
-    }
+        regions = topologyConfig.getFloodlightRegions().stream()
+                .filter(entry -> !Strings.isNullOrEmpty(entry))
+                .collect(Collectors.toSet());
+        if (regions.isEmpty()) {
+            throw new ConfigurationException("regions list must not be empty");
+        }
 
-    private void createKildaFlowHsKafkaBolt(TopologyBuilder builder, int parallelism, KafkaTopicsConfig topicsConfig) {
-        KafkaBolt kildaFlowHsKafkaBolt = buildKafkaBoltWithAbstractMessageSupport(topicsConfig.getFlowHsSpeakerTopic());
-        builder.setBolt(ComponentType.KILDA_FLOW_HS_KAFKA_BOLT, kildaFlowHsKafkaBolt, parallelism)
-                .shuffleGrouping(ComponentType.KILDA_FLOW_HS_REPLY_BOLT, Stream.KILDA_HS_FLOW);
+        kafkaTopics = topologyConfig.getKafkaTopics();
+        persistenceManager = PersistenceProvider.getInstance().createPersistenceManager(configurationProvider);
     }
 
-    private void createKildaFlowHsReplyStream(TopologyBuilder builder, int spoutParallelism, int parallelism,
-                                              KafkaTopicsConfig topicsConfig, List<String> kildaFlowHsTopics) {
-        createKildaFlowHsSpout(builder, spoutParallelism, kildaFlowHsTopics);
-        createKildaFlowHsKafkaBolt(builder, parallelism, topicsConfig);
+    @Override
+    public StormTopology createTopology() {
+        logger.info("Creating FlowTopology - {}", topologyName);
```

Comment:
Flow -> Floodlightrouter

Reply:
+1

Reply:
Weird log message in a weird place - it works during topology uploading. 

PS It is "FlowTopology" in our main branch now. :)